Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelperTest.java

```diff
@@ -0,0 +1,219 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package gov.hhs.fha.nhinc.transform.policy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import gov.hhs.fha.nhinc.common.eventcommon.XDREventType;
+import gov.hhs.fha.nhinc.common.eventcommon.XDRMessageType;
+import gov.hhs.fha.nhinc.common.eventcommon.XDRResponseEventType;
+import gov.hhs.fha.nhinc.common.eventcommon.XDRResponseMessageType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.HomeCommunityType;
+import gov.hhs.fha.nhinc.common.nhinccommon.PersonNameType;
+import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestType;
+import ihe.iti.xds_b._2007.ProvideAndRegisterDocumentSetRequestType;
+import ihe.iti.xds_b._2007.ProvideAndRegisterDocumentSetRequestType.Document;
+import oasis.names.tc.ebxml_regrep.xsd.lcm._3.SubmitObjectsRequest;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectListType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotListType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ValueListType;
+import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryResponseType;
+import oasis.names.tc.xacml._2_0.context.schema.os.AttributeType;
+import oasis.names.tc.xacml._2_0.context.schema.os.RequestType;
+import oasis.names.tc.xacml._2_0.context.schema.os.SubjectType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class XDRPolicyTransformHelperTest {
+
+	private static final String MESSAGE_ID = "12345566";
+	private XDRPolicyTransformHelper xdrTransformHelper;
+	private HomeCommunityType hct;
+	private AssertionType assertionType;
+	private SlotType1 slotType1;
+	private ValueListType valueListType;
+	private PersonNameType pnt;
+	private SlotListType rsl;
+	private RegistryObjectListType rolt;
+	private SubmitObjectsRequest sor;
+	private static final String TEST_HC_VAL = "urn:oid:1.1.1.1.1.1.1";
+
+	public XDRPolicyTransformHelperTest() {
+	}
+
+	@Before
+	public void setUp() {
+		xdrTransformHelper = new XDRPolicyTransformHelper();
+		hct = new HomeCommunityType();
+		hct.setHomeCommunityId(TEST_HC_VAL);
+		hct.setName("hct_Name");
+		hct.setDescription("hct_description");
+		assertionType = new AssertionType();
+		slotType1 = new SlotType1();
+		valueListType = new ValueListType();
+		pnt = new PersonNameType();
+		rsl = new SlotListType();
+		rolt = new RegistryObjectListType();
+		sor = new SubmitObjectsRequest();
+
+	}
+
+	@After
+	public void tearDown() {
+		xdrTransformHelper = null;
+	}
+
+	@Test
+	public void testTransformXDRToCheckPolicy() throws Exception {
+
+		XDREventType eventA = new XDREventType();
+
+		assertionType.setMessageId(MESSAGE_ID);
+
+		slotType1.setName("$XDSDocumentEntryPatientId");
```

Comment:
I want to keep it same manner... In AdhocQueryPolicyTest, was was done which Jason, Minh and you approved...